src/networking.c

```diff
@@ -3276,14 +3276,14 @@ sds catClientInfoString(sds s, client *client) {
         replBufBlock *cur = listNodeValue(client->ref_repl_buf_node);
         used_blocks_of_repl_buf = last->id - cur->id + 1;
     }
-
+    
     /* clang-format off */
     sds ret = sdscatfmt(s, FMTARGS(
         "id=%U", (unsigned long long) client->id,
         " addr=%s", getClientPeerId(client),
         " laddr=%s", getClientSockname(client),
         " %s", connGetInfo(client->conn, conninfo, sizeof(conninfo)),
-        " name=%s", client->name ? (char*)client->name->ptr : "",
+        " name=%s", server.hide_user_data_from_log ? "*redacted*" : (client->name ? (char*)client->name->ptr : ""),
```

Comment:
I am not sure if we should explicitly say there: the part is redacted.  Maybe empty string "" is enough

Reply:
I think that an empty string could be ambiguous and might make users think that the client's name is an empty string. By explicitly writing "*redacted*", it becomes clear that the user data has been intentionally hidden or redacted from the log.